skimage/io/sift.py

```diff
@@ -44,8 +38,13 @@ def _sift_read(f, mode='SIFT'):
           feature values
 
     """
-    if not hasattr(f, 'readline'):
+    if isinstance(f, str):
         f = open(f, 'r')
+        f_is_str = True
+    elif hasattr(f, 'readline'):
```

Comment:
This logic seems overly complicated; just check for string type, otherwise try to open as file.